### PR TITLE
tkt-70327: ix-snmpd to middlewared

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: src/freenas/etc/ix.rc.d/ix-snmpd [hunk 1]
```diff
@@ -1,72 +0,0 @@
...
-				if [ -n "${snmp_v3_privproto}" -a -n "${snmp_v3_privpassphrase}" ]; then
-					echo " ${snmp_v3_privproto} \"${snmp_v3_privpassphrase}\"" >> ${snmp_config}
-				else
-					echo >> ${snmp_config}
-				fi
-				echo "rwuser ${snmp_v3_username}" >> ${snmp_config}
-			fi
-
-		else
-			echo "rocommunity \"${snmp_community}\" default" >> ${snmp_config}
```
Comment (sonicaj) [inline src/freenas/etc/ix.rc.d/ix-snmpd]: We are missing these in the new conf file ? Is that intentional ?

File: src/middlewared/middlewared/etc_files/local/snmpd.conf [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+agentAddress udp:161,udp6:161,unix:/var/run/snmpd.sock
+sysLocation ${config["location"] or "unknown"}
+sysContact ${config["contact"] or "unknown@localhost"}
+sysDescr ${freenas_version}. Hardware: ${hw_machine} ${hw_model} running at ${hw_clockrate}. Software: ${kern_ostype} ${kern_osrelease} (revision ${kern_osrevision})
+
+master agentx
+
+% if config["v3"]:
+    % if config["v3_username"] and config["v3_password"]:
+createUser ${config["v3_username"]} ${config["v3_authtype"]} "${config["v3_password"]}" \
```
Comment (sonicaj) [inline src/middlewared/middlewared/etc_files/local/snmpd.conf]: We have a trailing `\`

File: src/freenas/etc/ix.rc.d/ix-snmpd (hunk 1 above)
Comment (themylogin) [reply src/freenas/etc/ix.rc.d/ix-snmpd]: Good catch!

File: src/middlewared/middlewared/etc_files/local/snmpd.conf (hunk 2 above)
Comment (themylogin) [reply src/middlewared/middlewared/etc_files/local/snmpd.conf]: This is intentional so `${config["v3_privproto"]} "${config["v3_privpassphrase"]}"` stays on the same line

File: src/freenas/etc/ix.rc.d/ix-snmpd (hunk 1 above)
Comment (themylogin) [reply src/freenas/etc/ix.rc.d/ix-snmpd]: Fixed

File: src/middlewared/middlewared/etc_files/local/snmpd.conf (hunk 2 above)
Comment (sonicaj) [reply src/middlewared/middlewared/etc_files/local/snmpd.conf]: Oh yup, missed that. Thank you for pointing it out

File: src/middlewared/middlewared/etc_files/local/snmpd.conf [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<%
+    import subprocess
+
+    hw_machine = subprocess.check_output("/sbin/sysctl -n hw.machine", shell=True, encoding="utf-8").strip()
+    hw_model = subprocess.check_output("/sbin/sysctl -n hw.model", shell=True, encoding="utf-8").strip()
+    hw_clockrate = subprocess.check_output("/sbin/sysctl -n hw.clockrate", shell=True, encoding="utf-8").strip()
+    kern_ostype = subprocess.check_output("/sbin/sysctl -n kern.ostype", shell=True, encoding="utf-8").strip()
+    kern_osrelease = subprocess.check_output("/sbin/sysctl -n kern.osrelease", shell=True, encoding="utf-8").strip()
+    kern_osrevision = subprocess.check_output("/sbin/sysctl -n kern.osrevision", shell=True, encoding="utf-8").strip()
```
Comment (william-gr) [inline src/middlewared/middlewared/etc_files/local/snmpd.conf]: We should use sysctl module here, that would avoid a lot of process calls

File: src/middlewared/middlewared/etc_files/local/snmpd.conf [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    freenas_version = subprocess.check_output("cat /etc/version", shell=True, encoding="utf-8").strip()
```
Comment (william-gr) [inline src/middlewared/middlewared/etc_files/local/snmpd.conf]: system.info would make more sense here, or open()

**Pull request merged**